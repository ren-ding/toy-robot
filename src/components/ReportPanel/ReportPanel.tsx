import React, { Component } from 'react';
import './styles/ReportPanel.css'
const uuidv4 = require('uuid/v4');

interface Props {
  results:Array<string>
}

export default class ReportPanel extends Component<Props> {
  constructor(props:Props) {
    super(props);
  }
  
  render() {
    return (
      <div id='report-panel-wrapper' className='report-panel-wrapper'>
        <span className='report-panel-label'>Results:</span>
        <div id='report-panel-results' className='report-panel-results'>
          {this.props.results.map((value,index)=> (<div key={uuidv4()}>{value}</div>))}
        </div>
      </div>
    );
  }
}