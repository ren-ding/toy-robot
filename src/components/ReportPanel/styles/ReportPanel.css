.report-panel-wrapper
{
    flex: 1;
    display: flex;
    flex-direction: column;
    margin: 20px;
}

.report-panel-wrapper .report-panel-label
{
    font-size: 25px;
    margin: 5px;
}

.report-panel-wrapper .report-panel-results
{
    height: 460px;
    width: 300px;
    font-size: 20px;
    border-style: solid;
    border-radius: 5px;
    border-color: darkgray;
    border-width: 1px;
    padding: 2px;
}
