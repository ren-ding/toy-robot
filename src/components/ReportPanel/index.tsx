import ReportPanel from './ReportPanel'
export default ReportPanel