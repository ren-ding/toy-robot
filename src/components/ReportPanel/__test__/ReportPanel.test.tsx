import React from 'react';
import ReportPanel from '../ReportPanel'
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

interface Props {
  results:Array<string>
}

describe('ReportPanel', ()=>{
  let reportPanel:any;
  let props: Props;

  beforeEach(()=>{
    props = {
      results:[]
    };

    reportPanel = () =>shallow(<ReportPanel {...props} />, { lifecycleExperimental: true })
  });

  describe('initialize with an empty string results props', ()=>{
    it('should render a title span and an empty textarea', () => {
      const panel = reportPanel();
      expect(panel.html()).toContain('Results:');
      expect(panel.find('#report-panel-results').length).toBe(1);
      expect(panel.find('#report-panel-results').text()).toBe('');
    });
  });
  
  describe('initialize with results', ()=>{
    it('should render a title span and the textarea with results', () => {
      const results = ['3,3,NORTH','3,4,NORTH','3,4,WEST'];
      const textResults = results.reduce((accumulator,currentValue)=> accumulator + currentValue);
      props.results = results;
      const panel = reportPanel();
      expect(panel.html()).toContain('Results:');
      expect(panel.find('#report-panel-results').length).toBe(1);
      expect(reportPanel().find('#report-panel-results').text()).toBe(textResults);      
    });
  });
  
});