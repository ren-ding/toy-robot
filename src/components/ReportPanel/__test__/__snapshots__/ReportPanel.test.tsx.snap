// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportPanel initialize with an empty string results props should render to match with snapshot 1`] = `
<div
  className="report-panel-wrapper"
  id="report-panel-wrapper"
>
  <span
    className="report-panel-label"
  >
    Results:
  </span>
  <div
    className="report-panel-results"
    id="report-panel-results"
  />
</div>
`;
